project/MergeFilesStrategy.scala

```diff
@@ -0,0 +1,22 @@
+import java.io.ByteArrayOutputStream
+import java.nio.file.Files
+
+import org.apache.logging.log4j.core.config.plugins.processor.PluginCache
+import sbt.File
+import sbtassembly.MergeStrategy
+
+import scala.collection.JavaConverters.asJavaEnumerationConverter
+
+class MergeFilesStrategy extends MergeStrategy {
+  override def name: String = "mergeFilesStrategy"
+
+  override def apply(tempDir: File, path: String, files: Seq[File]): Either[String, Seq[(File, String)]] = {
```

Comment:
what's that for?

Reply:
log4j2 plugin caches are needed for loading appenders. Amazon AWS Lambda log4j2 support provides it's own.
When sbt assembly is run this merges the files instead of attempting to guess at which to discard which can result in breaking logging.
Renaming to MergeLog4j2PluginCachesStrategy.